fabtests/pytest/efa/efa_common.py

```diff
@@ -122,3 +128,79 @@ def get_efa_device_names(server_id):
         assert itemlist[1] in ["rdm", "dgrm"]
         device_names.add(itemlist[0])
     return list(device_names)
+
+
+class EFAMultinodeClientServerTest:
+    """
+    This class establishes tests involving a single server process
+    and multiple client processes which can be initiated on multiple
+    nodes (configured via client_host_name_list).
+    """
+
+    def __init__(self, cmdline_args, base_command, client_hostname_list):
+        self._cmdline_args = cmdline_args
+        self._base_command = base_command
+        self._timeout = self._cmdline_args.timeout
+        self.client_hostname_list = client_hostname_list
+        self.numclient = len(client_hostname_list)
+        self._server_command = cmdline_args.populate_command(self._base_command + " -C {}".format(self.numclient), "server", self._timeout)
+        assert(isinstance(client_hostname_list, list) and len(client_hostname_list) >= 1)
```

Comment:
Move L147 before L141. Input validation should happen earlier. Otherwise the program could panic on L145 if `client_hostname_list=None`.